omc3/scripts/luminosity_imbalance.py

```diff
@@ -0,0 +1,142 @@
+from math import sqrt
+from functools import reduce
+from operator import mul
+
+import tfs
+from tfs.tools import significant_digits
+
+from omc3.utils.logging_tools import get_logger
+from omc3.definitions.constants import PLANES
+from generic_parser import entrypoint, EntryPointParameters
+
+LOG = get_logger(__name__)
+
+BETASTAR = 'BETSTAR'
+ERR = 'ERR'
+LABEL = 'LABEL'  # Column containing the IP/Beam names
+BEAMS = ('B1', 'B2')
+IPS = ('IP1', 'IP5')
+LABELS = [f'{i}{b}' for i in IPS for b in BEAMS]
+
+
+def get_params():
+    return EntryPointParameters(
+            tfs=dict(
+                required=True,
+                type=str,
+                help="TFS file containing the β* and error for "
+                     "B{1,2}{H,V}IP{1,5}"
+            ),
+            inplace=dict(
+                required=False,
+                action='store_true',
+                help="Add the luminosity imbalance in the source TFS file as a"
+                     "header"
+            )
+    )
+
+
+def _validate_tfs(df):
+    # First validate the labels, which correspond to the IP and Beam
+    expected_labels = set(LABELS)
+    if len(df[LABEL]) != len(LABELS) or set(df[LABEL]) != expected_labels:
```

Comment:
Actually, now that I think about it... you should allow other labels as well, as you are filtering by IP anyway. In case k-mod data includes other IPs. So your check would be like below:
```
not_found_labels = [l for l in expected_labels if l not in df[LABEL]]
if any(not_found_labels):
    msg = f'The following required labels are not found in dataframe: {str(not_found_labels).strip("[]")}`
```

Reply:
`list2str` from before can be used to shorten